packages/x-charts/src/ChartsLabel/ChartsLabelGradient.test.tsx

```diff
@@ -28,6 +31,163 @@ describe('<ChartsLabelGradient />', () => {
   }));
+
+  // JSDOM does not support SVGMatrix
+  describeSkipIf(isJSDOM)('rotation', () => {
+    const matrixToRotation = (element: SVGSVGElement | null) => {
+      if (!element) {
+        throw new Error('Svg element not found');
+      }
+
+      const matrix = new DOMMatrix(getComputedStyle(element).transform);
+      return (Math.atan2(matrix.b, matrix.a) * 180) / Math.PI;
+    };
+
+    describe('horizontal', () => {
+      it('should render a gradient in the correct orientation', () => {
+        const { container } = render(<ChartsLabelGradient gradientId="gradient.test-id" />);
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(0);
+      });
+
+      it('should reverse the gradient', () => {
+        const { container } = render(<ChartsLabelGradient gradientId="gradient.test-id" reverse />);
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(180);
+      });
+
+      it('should rotate the gradient', () => {
+        const { container } = render(<ChartsLabelGradient gradientId="gradient.test-id" rotate />);
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(90);
+      });
+
+      it('should reverse and rotate the gradient', () => {
+        const { container } = render(
+          <ChartsLabelGradient gradientId="gradient.test-id" reverse rotate />,
+        );
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(-90);
+      });
+    });
+
+    describe('vertical', () => {
+      it('should render a gradient in the correct orientation', () => {
+        const { container } = render(
+          <ChartsLabelGradient gradientId="gradient.test-id" direction="vertical" />,
+        );
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(-90);
+      });
+
+      it('should reverse the gradient', () => {
+        const { container } = render(
+          <ChartsLabelGradient gradientId="gradient.test-id" direction="vertical" reverse />,
+        );
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(90);
+      });
+
+      it('should rotate the gradient', () => {
+        const { container } = render(
+          <ChartsLabelGradient gradientId="gradient.test-id" direction="vertical" rotate />,
+        );
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(0);
+      });
+
+      it('should reverse and rotate the gradient', () => {
+        const { container } = render(
+          <ChartsLabelGradient gradientId="gradient.test-id" direction="vertical" reverse rotate />,
+        );
+        const svg = container.querySelector('svg');
+        expect(matrixToRotation(svg)).to.equal(180);
+      });
+    });
+
+    describe('RTL', () => {
```

Comment:
Nice to have the RTL tested :+1: The standard is a bonus because most of the usecases are already tested by argos in https://deploy-preview-16115--material-ui-x.netlify.app/x/react-charts/legend/#label-position